### PR TITLE
Move GetStatefulSet, BuildWorkAffinity as common function

### LINKED ISSUE #1064: [BUG]Move GetStatefulSet as common function
**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅱ. Does this pull request fix one issue?

fixes #1064 

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ctrl/affinity.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			}
+		}
+
+		// 3. Perefer to locate on the node which already has fuse on it
```
Comment (yangyuliufeng) [inline pkg/ctrl/affinity.go:101]: prefer

Comment (cheyang) [reply pkg/ctrl/affinity.go:101]: > prefer

Thanks

**Review by yangyuliufeng — COMMENTED** [review]
It's a good idea to put the common part of all engine into a special struct. But a think it ( the Helper) should be placed in pkg/ddc/common instead of pkg/ctrl. Also, the go file should not call affinity.go because we may put more common functions into it.

**Comment by cheyang** [comment]
> It's a good idea to put the common part of all engine into a special struct. But a think it ( the Helper) should be placed in pkg/ddc/common instead of pkg/ctrl. Also, the go file should not call affinity.go because we may put more common functions into it.

I've think about it. The concern is `Circular dependency`. In common pkg, we define some common attributes which are used by other package.

**Review by yangyuliufeng — APPROVED** [review]
LGTM

**Comment by cheyang** [comment]
thanks, @yangyuliufeng

**Pull request merged**